lib/OpenQA/Worker/Job.pm

```diff
@@ -244,6 +244,10 @@ sub start {
     # set base dir to the one assigned with web UI
     $ENV{OPENQA_SHAREDIR} = $client->working_directory;
 
+    # Store the packages list where runs the worker
+    my $packages_file = $worker->pool_directory . '/worker_packages';
+    system($global_worker_settings->{'PACKAGES_CMD'} . ' >' . $worker->pool_directory . '/worker_packages.txt');
```

Comment:
Better store `$worker->pool_directory` in a variable (instead of `$packages_file` which looks unused. The code should also handle the case when `PACKAGES_CMD` is empty. I suppose the path should better be quoted. It would also make sense to log the execution of the command (e.g. it might hang and then it would be useful to deduce from the worker log what's taking so long).

```suggestion
    if (my $cmd = $global_worker_settings->{PACKAGES_CMD}) {
        my $pool_directory = $worker->pool_directory;
        log_info("Gathering package information: $cmd");
        system("$cmd > '$pool_directory/worker_packages.txt'");
    }
```